<!-- -*- mode: XML -*- -->
<launch>
  <arg name="node_name" default="boat_robot" />
  <!-- Use udev rules to setup aliases for right/left wheel port.
  <arg name="right_wheel_port" default="/dev/right_wheel_esc" />
  <arg name="left_wheel_port" default="/dev/right_wheel_esc" />
-->
  <arg name="right_thruster_port" default="/dev/ttyACM0" />
  <arg name="left_thruster_port" default="/dev/ttyACM1" />


  <!-- Optionally launch manager in GDB, for debugging -->
  <arg name="debug" default="false" />
  <arg if="$(arg debug)" name="launch_prefix" value="xterm -e gdb --args" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <!-- Also globally disable bond heartbeat timeout in debug mode, so everything
       doesn't die when you hit a break point -->
  <param if="$(arg debug)" name="/bond_disable_heartbeat_timeout" value="true" />

    <node pkg="boat_hw" type="robot" name="$(arg node_name)" output="screen" launch-prefix="$(arg launch_prefix)" >
      <param name="right_thruster_port" value="$(arg right_thruster_port)" />
      <param name="left_thruster_port" value="$(arg left_thruster_port)" />
      <param name="max_rpm" value="3250" type="double" />
      <param name="min_rpm" value="-3250" type="double" />
	<param name="throttle_smoother_rate" value="100" />
	<param name="rpm_to_erpm_gain" value="7" />	
    </node>
</launch>
